lib/adapters/tool/playwright/index.ts

```diff
@@ -0,0 +1,311 @@
+import type {EventSource} from '../../../gui/event-source';
+import type {PwtRawTest} from '../../test/playwright';
+import type {ReporterConfig} from '../../../types';
+import type {TestSpec} from '../types';
+import type {FullConfig, TestCase, TestResult} from '@playwright/test/reporter';
+import type {TestBranch} from '../../../tests-tree-builder/gui';
+import type {PwtEventMessage} from './reporter';
+
+export const DEFAULT_CONFIG_PATHS = [
+    `${ToolName.Playwright}.config.ts`,
+    `${ToolName.Playwright}.config.js`,
+    `${ToolName.Playwright}.config.mts`,
+    `${ToolName.Playwright}.config.mjs`,
+    `${ToolName.Playwright}.config.cts`,
+    `${ToolName.Playwright}.config.cjs`
+];
+
+export class PlaywrightToolAdapter implements ToolAdapter {
+    private _toolName: ToolName;
+    private _configPath: string;
+    private _config: PlaywrightConfigAdapter;
+    private _reporterConfig: ReporterConfig;
+    private _haveProjectsInConfig: boolean;
+    private _htmlReporter: HtmlReporter;
+    private _pwtBinaryPath: string;
+    private _reportBuilder!: GuiReportBuilder;
+    private _eventSource!: EventSource;
+    private _guiApi?: GuiApi;
+
+    static create(
+        this: new (options: ToolAdapterOptionsFromCli) => PlaywrightToolAdapter,
+        options: ToolAdapterOptionsFromCli
+    ): PlaywrightToolAdapter {
+        return new this(options);
+    }
+
+    constructor(opts: ToolAdapterOptionsFromCli) {
+        const {config, configPath} = readPwtConfig(opts);
+
+        this._configPath = configPath;
+        this._config = PlaywrightConfigAdapter.create(config);
+        this._toolName = opts.toolName;
+
+        const pluginOpts = getPluginOptions(this._config.original);
+        this._reporterConfig = parseConfig(pluginOpts);
+
+        this._haveProjectsInConfig = !_.isEmpty(this._config.original.projects);
+
+        this._htmlReporter = HtmlReporter.create(this._reporterConfig, {toolName: ToolName.Playwright});
+        this._pwtBinaryPath = npmWhich.sync(ToolName.Playwright, {cwd: process.cwd()});
+    }
+
+    get configPath(): string {
+        return this._configPath;
+    }
+
+    get toolName(): ToolName {
+        return this._toolName;
+    }
+
+    get config(): PlaywrightConfigAdapter {
+        return this._config;
+    }
+
+    get reporterConfig(): ReporterConfig {
+        return this._reporterConfig;
+    }
+
+    get htmlReporter(): HtmlReporter {
+        return this._htmlReporter;
+    }
+
+    get guiApi(): GuiApi | undefined {
+        return this._guiApi;
+    }
+
+    initGuiApi(): void {
+        this._guiApi = GuiApi.create();
+    }
+
+    async readTests(): Promise<PlaywrightTestCollectionAdapter> {
+        const stdout = await new Promise<string>((resolve, reject) => {
+            // specify default browser in order to get correct stdout with browser name
+            const browserArgs = this._haveProjectsInConfig ? [] : ['--browser', DEFAULT_BROWSER_ID];
```

Comment:
I need to specify `chromium` if user not use `projects` in order to get correct stdout with browser name